pallets/inactivity-tracking/src/lib.rs

```diff
@@ -154,6 +156,11 @@ pub mod pallet {
     pub type ActiveCollatorsForCurrentSession<T: Config> =
         StorageValue<_, BoundedBTreeSet<T::CollatorId, T::MaxCollatorsPerSession>, ValueQuery>;
 
+    /// A list of active container chains for a session. Repopulated at the start of every session
+    #[pallet::storage]
+    pub type ActiveContainerChainsForCurrentSession<T: Config> =
+        StorageValue<_, BoundedBTreeSet<ParaId, T::MaxContainerChains>, ValueQuery>;
```

Comment:
So this starts empty, then using `on_chain_noted` active chains are inserted here, and then when the session ends it is "processed" and then cleared again.

The processing filters inactive collators. If a collator was assigned to an inactive chain it is not considered inactive.

When we add parathreads it is possible for one collator to be assigned to more than one chain, what happens in that case? If one chain is inactive but the other chain not? Or maybe this mark inactive feature doesn't make sense with parathreads.

Reply:
Very good point! If there is a possibility for a collator to produce blocks on multiple chain perhaps we can change how `process_inactive_chains_for_current_session`  work.
1) Iterate over all chains an create 2 vectors of collators - one for active chains and one for inactive.
2) Iterate over all inactive collators and add them to `ActiveCollatorsForCurrentSession` only if they 
	- aren't already there
	- aren't in the active chains collators vector
 This should solve the issue

Reply:
Added additional check that will prevent parathreads from being processed as inactive parachains for now in https://github.com/moondance-labs/tanssi/pull/899/commits/a3869cb10695bf6dc60214f54ea0bb63776fbf94 .